swap/src/protocol/alice/refund.rs

```diff
@@ -0,0 +1,123 @@
+use crate::bitcoin::{self};
+use crate::database::{Database, Swap};
+use crate::monero;
+use crate::protocol::alice::AliceState;
+use anyhow::{bail, Result};
+use libp2p::PeerId;
+use std::sync::Arc;
+use uuid::Uuid;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    // Errors indicating the the swap can *currently* not be refunded but might be later
+    #[error("Cannot refund because swap is not in a cancelled state. Make sure to cancel the swap before trying to refund or use --force.")]
```

Comment:
This error message knows too much about its context. Think about how this will be composed:

A function `fn refund() -> Result<_, Error>` can be called with: `refund().context("Failed to refund swap")?`;

What this error needs to say is **why** it failed to refund. The fact that it failed is already implied by the error being instantiated.

If you compose these errors together, what you want to see is:

```
Failed to refund swap.

Caused by:
	Swap is not in a cancelled state
```

Your current implementation will say:


```
Failed to refund swap.

Caused by:
	Cannot refund because swap is not in a cancelled state.
```

Which is redundant.

Reply:
Another hint that could have told you this is that you are repeating yourself at the beginning of every error message in this enum.